src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -975,7 +984,7 @@ public override BoundNode VisitLocalDeclaration(BoundLocalDeclaration node)
             (initializer, conversion) = RemoveConversion(initializer, includeExplicitConversions: false);
 
             TypeSymbolWithAnnotations valueType = VisitRvalueWithResult(initializer);
-            TypeSymbolWithAnnotations type = local.Type;
+            TypeSymbolWithAnnotations type = GetAdjustedType(local.Type);
```

Comment:
>GetAdjustedType [](start = 45, length = 15)

I see that parameters and locals are covered. Can you also check that `out` variables and other variables (using, foreach, deconstruction, ...) behave correctly?
The `let` in LINQ also introduces variables, but LINQ is known to be unhandled at the moment.

Reply:
Hum, ignore foreach variables for now. They have bigger problems.